src/dodal/devices/undulator.py

```diff
@@ -63,3 +90,49 @@ def __init__(
                 self.length = None
 
         super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def set(self, value: float):
```

Comment:
Should: This needs a docstring to make it clear you are setting energy and in what units, otherwise people may think you should be setting a gap here